ДЗ-3

PleasyShell opened the pull request.

PleasyShell committed:
Squashed commit of the following:

commit de3752ff3c162896a4a8d153332b7f95a044e4dc
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Tue Apr 16 10:27:03 2024 +0400

    Добавил проверку на пустой массив задач

    Ранее без проверки если отсутствовали элементы в массиве задач и добавлялись новые, то вместо id выдавалось infinity. Проблема устранена.

commit 3c61abd9753a76b200975cf1e268518a62ea5e59
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Tue Apr 16 10:18:18 2024 +0400

    Разбил добавление задачи на 2 функции

commit 63b73d9dfec92a61621ab795923969902b0804fc
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Tue Apr 16 09:49:27 2024 +0400

    Сделал все кнопки как компоненты

commit 64dc0f633b72533a2874e400146eae8ae895e3d8
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Tue Apr 16 08:41:31 2024 +0400

    ДЗ-3 Добавил спиннер загрузки списка задач

commit 782ab9f3deb6efc21073d33cb2a0f41f98348556
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Tue Apr 16 08:20:34 2024 +0400

    ДЗ-3 Добавил компонент кнопки

commit 44195b4a9bed3b3a914cd0ae4afcb4012b09aeba
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Mon Apr 15 09:24:43 2024 +0400

    Мелкий фикс

    Указал неверный путь к стилям в импорте связующего файла.

commit 82172dd32281ed11da3cefad112d08e5f9bda2bb
Author: Darklord Inc <avorobev341@gmail.com>
Date:   Mon Apr 15 08:26:27 2024 +0400

    Добавил связующий файл стилей

    Чтобы устранить длинный путь в файлам стилей.

PleasyShell commented on src/app/module/content/todolist-content/todolist-form/todolist-form.component.ts:
```diff
@@ -1,28 +1,37 @@
+        }, 500)
+    };
+
+    protected CheckTaskValid(value: string) {
+
+        value !== '' ? this.SaveTask(value) : console.log('Поле пустое!');
+    };
+
```
Разделил метод добавление на 2 метода. Сначала проверяется условие не пусто ли значение в инпуте создания задачи. Если нет, то перейти к сохранению задачи, иначе вывести в консоле, что поле пустое.

PleasyShell commented on src/app/module/content/todolist-content/todolist-form/todolist-form.component.ts:
```diff
@@ -1,28 +1,37 @@
+        value !== '' ? this.SaveTask(value) : console.log('Поле пустое!');
+    };
+
+    private SaveTask(taskname: string) {
```
В методе добавление сделал проверку, которая проверяет на пустой массив. Если он пуст, то новый элемент добавленной задачи начнет отсчет с 1. Ранее если элементы в массиве отсутствовали, то в id выдавалось infinity.

PleasyShell commented on src/app/module/content/todolist-content/todolist.component.ts:
```diff
@@ -3,7 +3,7 @@ import { Component } from '@angular/core';
   selector: 'todolist-content',
   templateUrl: './todolist.component.html',
-  styleUrls: ['../../../styles/pages-styles.scss']
+  styleUrls: ['./core.scss']
```
Связующий файл стилей core.scss я создал ради того, чтобы уменьшить и так длинный путь к стилям компоненту.

koshelnikov commented on src/app/module/components/buttons/delete-button/delete-button.component.html:
```diff
@@ -0,0 +1 @@
+<img class="delete-button" src="assets/images/Иконка-Удалить.svg">
```
используйте везде латиницу images/icon-remove.svg например так, или положите в папку images/icons/remove.svg

PleasyShell replied:
Изменю.

koshelnikov commented on src/app/module/components/buttons/save-button/save-button.component.html:
```diff
@@ -0,0 +1,3 @@
+<button class="save-button" [ngClass]="{'disabled': isDisable}">
+    <p class="text-in-buttons">Добавить</p>
```
save-button
save-button-text

Но тут другая проблема есть :) b>p нельзя использовать. Внутри кнопок используйте inline элементы, например span

https://caninclude.glitch.me/caninclude?child=p&parent=button

PleasyShell replied:
Хорошо) Насчет класса text-in-buttons, у меня глобален именно для кнопок. Т.е текст в любой кнопке имеет этот класс(стилизацию), поэтому я его так назвал.

koshelnikov commented on src/app/module/components/buttons/save-button/save-button.component.ts:
```diff
@@ -0,0 +1,13 @@
+})
+export class SaveButtonComponent {
+
+  @Input() Title: string = '';
```
title с маленькой буквы

PleasyShell replied:
Да, уже изменил, спасибо.

PleasyShell committed:
Именования свойств и методов со строчной буквы

Сделал именования свойств и методов со строчной буквы

PleasyShell committed:
Перевёл глобальный файл стилей проекта в формат scss

Т.к я создавал через шаблон VS, а не терминал, у меня автоматические создается всегда в формате css.

PleasyShell committed:
Cделал сервис закрытым к данным массива задач.

Сделал сервис закрытым, методы взаимодействия с задачи вынес в сервис. Получение и манипуляции с данными идут вызываются с компонентов, но сами действия с данными происходят непосредственно в сервисе задач.

PleasyShell committed:
Перераспределил стили проекта
